### PR TITLE
Fixed clang build warnings.

### PR DESCRIPTION
**Pull request by RICCIARDI-Adrien**
…ing.

### CODE REVIEW

**Comment by RICCIARDI-Adrien** [comment]
All clang warnings have been fixed. This pull request is ready to be merged.

**Comment by Gh0stBlade** [comment]
Good work.

File: src/core/console.c [hunk 1]
```diff
@@ -68,10 +68,10 @@ void Con_SetExecFunction(int(*exec_cmd)(char *ch))
 
 void Con_InitGlobals()
 {
-    con_base.background_color[0] = 1.0f;
-    con_base.background_color[1] = 0.9f;
-    con_base.background_color[2] = 0.7f;
-    con_base.background_color[3] = 0.4f;
+    con_base.background_color[0] = 0;
```
Comment (Gh0stBlade) [inline src/core/console.c]: Shouldn't this be 255?

File: src/core/console.c [hunk 2]
```diff
@@ -68,10 +68,10 @@ void Con_SetExecFunction(int(*exec_cmd)(char *ch))
...
+    con_base.background_color[1] = 0;
```
Comment (Gh0stBlade) [inline src/core/console.c]: Shouldn't this be 230?

File: src/core/console.c [hunk 3]
```diff
@@ -68,10 +68,10 @@ void Con_SetExecFunction(int(*exec_cmd)(char *ch))
...
+    con_base.background_color[2] = 0;
```
Comment (Gh0stBlade) [inline src/core/console.c]: Shouldn't this be 179

File: src/core/console.c [hunk 4]
```diff
@@ -68,10 +68,10 @@ void Con_SetExecFunction(int(*exec_cmd)(char *ch))
...
+    con_base.background_color[3] = 200;
```
Comment (Gh0stBlade) [inline src/core/console.c]: Shouldn't this be 102?

File: src/gui/gui_inventory.cpp [hunk 5]
```diff
@@ -1120,7 +1120,7 @@ void gui_InventoryManager::handlePassport(struct base_item_s *bi, float time)
                 m_current_menu = NULL;
             }
             Anim_IncTime(&bi->bf->animations, time);
-            if((bi->bf->animations.frame_changing_state == SS_CHANGING_END_ANIM))
+            if(bi->bf->animations.frame_changing_state == SS_CHANGING_END_ANIM)
```
Comment (Gh0stBlade) [inline src/gui/gui_inventory.cpp]: Approved

File: src/gui/gui_inventory.cpp [hunk 6]
```diff
@@ -1197,7 +1197,7 @@ void gui_InventoryManager::handleCompass(struct base_item_s *bi, float time)
             m_current_menu = NULL;
         }
         Anim_IncTime(&bi->bf->animations, time);
-        if((bi->bf->animations.frame_changing_state == SS_CHANGING_END_ANIM))
+        if(bi->bf->animations.frame_changing_state == SS_CHANGING_END_ANIM)
```
Comment (Gh0stBlade) [inline src/gui/gui_inventory.cpp]: Approved.

File: src/state_control/state_control_Lara.cpp [hunk 7]
```diff
@@ -3549,10 +3549,10 @@ void StateControl_LaraSetWeaponModel(struct entity_s *ent, int weapon_model, int
                 else if (anim_th->current_animation == 0)
                 {
                     // check if weapon is not grenadegun to hide the weapon
-                    if (IS_TR_I(ver) && anim_th->model->id == TR1_MODEL_SHOTGUN ||
-                        IS_TR_II(ver) && anim_th->model->id != TR2_MODEL_GRENADEGUN ||
-                        IS_TR_III(ver) && anim_th->model->id != TR3_MODEL_GRENADEGUN ||
-                        (ver >= TR_IV) && anim_th->model->id != TR4C_MODEL_GRENADEGUN
+                    if ((IS_TR_I(ver) && (anim_th->model->id == TR1_MODEL_SHOTGUN)) ||
```
Comment (Gh0stBlade) [inline src/state_control/state_control_Lara.cpp]: Approved

**Comment by RICCIARDI-Adrien** [comment]
Thank you !

File: src/engine.cpp [hunk 8]
```diff
@@ -155,7 +155,7 @@ void Engine_Start(int argc, char **argv)
     // Primary initialization.
     Engine_Init_Pre();
 
-    Engine_LoadConfig(config_name ? config_name : "config.lua");
+    Engine_LoadConfig(config_name);
```
Comment (Gh0stBlade) [inline src/engine.cpp]: Approved, also fixes bug where config_name is always true (it shouldn't be treated as pointer) maybe it is previously treated as pointer due to config_name previously being of type char*

File: src/core/console.c (hunk 1 above)
Comment (RICCIARDI-Adrien) [reply src/core/console.c]: I set the values that are currently in use in the repo `config.lua` file (`background_color = {r = 0, g = 0, b = 0, a = 200};`), but you can use the old color values if you prefer.

File: src/engine.cpp (hunk 8 above)
Comment (RICCIARDI-Adrien) [reply src/engine.cpp]: Yes, it previously was a pointer. I forgot to change this line in my previous commit c1bd68b76b4a7617e0c963fe76b87ad76445d1ad.

**Pull request merged**